src/main/java/com/google/devtools/build/lib/rules/cpp/CcBinary.java

```diff
@@ -430,7 +430,8 @@ public static ConfiguredTarget init(CppSemantics semantics, RuleContext ruleCont
     CcLinkingContext depsCcLinkingContext = collectCcLinkingContext(ruleContext);
 
     Artifact generatedDefFile = null;
-    Artifact customDefFile = null;
+    Artifact customDefFile;
+    Artifact winDefFile = null;
```

Comment:
`generatedDefFile` is used again in line 641. I wonder, do you have to use `winDefFile` there instead? If so, please update and remove `generatedDefFile` here and declare it in the `if` block's scope below.

Reply:
No, it should still be geenratedDefFile in line 641. Because that's for users to get the generated DEF file via `def_file` output group, it should work even when `windows_export_all_symbols` feature is not enabled.